src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+        return _totalStakeHistory[quorumNumber][_totalStakeHistory[quorumNumber].length - 1].stake;
+    }
+
+    function getLengthOfOperatorIdStakeHistoryForQuorum(bytes32 operatorId, uint8 quorumNumber) external view returns (uint256) {
+        return operatorIdToStakeHistory[operatorId][quorumNumber].length;
+    }
+
+    function getLengthOfTotalStakeHistoryForQuorum(uint8 quorumNumber) external view returns (uint256) {
+        return _totalStakeHistory[quorumNumber].length;
+    }
+
+    /**
+     * @notice Checks that the `operator` was active at the `blockNumber`, using the specified `stakeHistoryIndex` as proof.
+     * @param operatorId is the id of the operator of interest
+     * @param blockNumber is the block number of interest
+     * @param quorumNumber is the quorum number which the operator had stake in
+     * @param stakeHistoryIndex specifies an index in `operatorIdToStakeHistory[operatorId]`
+     * @return 'true' if it is succesfully proven that  the `operator` was active at the `blockNumber`, and 'false' otherwise
+     * @dev In order for this function to return 'true', the inputs must satisfy all of the following list:
+     * 1) `operatorIdToStakeHistory[operatorId][quorumNumber][index].updateBlockNumber <= blockNumber`
+     * 2) `operatorIdToStakeHistory[operatorId][quorumNumber][index].nextUpdateBlockNumber` must be either `0` (signifying no next update) or
+     * is must be strictly greater than `blockNumber`
+     * 3) `operatorIdToStakeHistory[operatorId][quorumNumber][index].stake > 0` i.e. the operator had nonzero stake
+     * @dev Note that a return value of 'false' does not guarantee that the `operator` was inactive at `blockNumber`, since a
+     * bad `stakeHistoryIndex` can be supplied in order to obtain a response of 'false'.
+     */
+    function checkOperatorActiveAtBlockNumber(
+        bytes32 operatorId,
+        uint256 blockNumber,
+        uint8 quorumNumber,
+        uint256 stakeHistoryIndex
+        ) external view returns (bool)
+    {
+        // pull the stake history entry specified by `stakeHistoryIndex`
+        OperatorStakeUpdate memory operatorStakeUpdate = operatorIdToStakeHistory[operatorId][quorumNumber][stakeHistoryIndex];
+        return (
+            // check that the update specified by `stakeHistoryIndex` occurred at or prior to `blockNumber`
+            (operatorStakeUpdate.updateBlockNumber <= blockNumber)
+            &&
+            // if there is a next update, then check that the next update occurred strictly after `blockNumber`
+            (operatorStakeUpdate.nextUpdateBlockNumber == 0 || operatorStakeUpdate.nextUpdateBlockNumber > blockNumber)
+            &&
+            /// verify that the stake was non-zero at the time (note: here was use the assumption that the operator was 'inactive'
+            /// once their stake fell to zero)
+            operatorStakeUpdate.stake != 0 // this implicitly checks that the operator was a part of the quorum of interest
```

Comment:
is this still a valid implicit assumption?

Reply:
bump on this. might mean just defining more clearly what `active` means?

Reply:
likewise for `checkOperatorInactiveAtBlockNumber`

Reply:
I think this might be broken, if there's a stake update to take the operator below a minimum stake their stake is set to 0 for no good reason, altered it here to update to 1. now this is a valid assumption: https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/4262ada9f18635b63937b546fa994e7964acb10b

Reply:
although that works, it feels a bit hacky -- could we instead explicitly check that the operator was/was not a member of the `quorumNumber`th quorum?

Reply:
ah great point. I'm fine taking these out, and later on (we need this during proofs of custody) adding functions to the registry coordinator?

Reply:
> ah great point. I'm fine taking these out, and later on (we need this during proofs of custody) adding functions to the registry coordinator?

that seems reasonable to me!

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/caab3f3f65cf22df01ebad02f283ebe1f44338c9